projects/packages/my-jetpack/_inc/data/purchases/use-purchases.ts

```diff
@@ -0,0 +1,34 @@
+import { useQuery } from '@tanstack/react-query';
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { REST_API_SITE_PURCHASES_ENDPOINT, REST_API_SITE_PURCHASES_QUERY_KEY } from '../constants';
+import useNotice from '../notices/use-notice';
+import type { Purchase } from '../../../global';
+import type { UseQueryResult } from '@tanstack/react-query';
+
+const cacheKey = REST_API_SITE_PURCHASES_QUERY_KEY;
+
+const usePurchases: () => UseQueryResult< Array< Purchase >, Error > = () => {
+	const queryResult = useQuery( {
+		queryKey: [ cacheKey ],
+		queryFn: async () =>
+			( await apiFetch( { path: REST_API_SITE_PURCHASES_ENDPOINT } ) ) as Array< Purchase >,
+		refetchOnWindowFocus: false,
+		refetchIntervalInBackground: false,
+	} );
+
+	const { isError } = queryResult;
+
+	useNotice( {
+		message: __(
+			'There was an error fetching your purchases information. Check your site connectivity and try again.',
+			'jetpack-my-jetpack'
+		),
+		options: { status: 'error' },
+		isError,
```

Comment:
Might be good to make sure it's not (re-)loading:
```suggestion
		! isLoading && isError,
```

Reply:
I agree, nice catch